src/components/Trans.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { memo } from "react";
+
+import { useI18n } from "../i18n";
+
+const REGEXP = /{{(.+?)}}/g;
+
+const getTransChildren = (format: string, props: { [key: string]: any }) => {
+  const stack: { name: string; children: any[] }[] = [
+    {
+      name: "",
+      children: [],
+    },
+  ];
+
+  format.split(REGEXP).forEach((match, index) => {
+    if (index % 2 === 0) {
+      if (match.length === 0) {
+        return;
+      }
+
+      stack[stack.length - 1].children.push(match);
+    } else if (match === `${stack[stack.length - 1].name}End`) {
+      const item = stack.pop()!;
+      const itemChildren =
+        item.children.length === 1 ? item.children[0] : item.children;
+      stack[stack.length - 1].children.push(props[item.name](itemChildren));
+    } else if (props.hasOwnProperty(match)) {
+      stack[stack.length - 1].children.push(props[match]);
+    } else if (/Start$/.test(match)) {
+      const name = match.slice(0, match.length - 5);
+
+      if (props.hasOwnProperty(name)) {
+        stack.push({
+          name,
+          children: [],
+        });
+      } else {
+        console.warn(
+          `getTransChildren: missed to pass in variable start/end ${match} for interpolating ${format}`,
+        );
+      }
+    } else {
+      console.warn(
+        `getTransChildren: missed to pass in variable ${match} for interpolating ${format}`,
+      );
+    }
+  });
+
+  if (stack.length !== 1) {
+    console.warn(
+      `getTransChildren: stack not empty for interpolating ${format}`,
+    );
+  }
+
+  return stack[0].children;
+};
+
+const Trans = memo<{ i18nKey: string; [key: string]: any }>(
+  ({ i18nKey, children, ...props }) => {
+    const { t, langCode } = useI18n();
+
+    const transChildren = React.useMemo(() => {
```

Comment:
The `useMemo` will never memoize due to `props`, and the component `memo` won't work either unless we memoized each render function pass to it which I doubt we'll do / would yield much benefit.

Reply:
It depends on the type of props. Props values can also be just strings (see example 1) but I agree that in the Excalidraw codebase, Trans props will probably be more complex so memoization won't make much sense.

Reply:
Removed `memo` and `useMemo` and added tests.

Reply:
> Props values can also be just strings (see example 1)

what would be the use case for using strings?

Reply:
Yes, `<span><Trans i18nKey="my.key" /></span>` is the same as `<span>{t("my.key")}</span>`.